crates/cheatcodes/src/utils.rs

```diff
@@ -145,6 +146,17 @@ impl Cheatcode for ensNamehashCall {
     }
 }
 
+impl Cheatcode for randomUintCall {
+    fn apply(&self, _state: &mut Cheatcodes) -> Result {
+        let Self {} = self;
+        // Use thread_rng to get a random number
+        let mut rng = rand::thread_rng();
```

Comment:
we could also store an rng instance that comes from config or something in the state, for seeding. cc @grandizzy

Reply:
Yea, I mentioned this in https://github.com/foundry-rs/foundry/issues/5407, but let's make sure using this cheat doesn't break determinism of fuzz runs when using a seed, or when replaying failures

Reply:
yep, I think we could draw values from a `cheatcode` strategy similar to `calldata` here
https://github.com/foundry-rs/foundry/blob/0a5b22f07ba4f2ddf525089c8ee9cdcb05e44bd9/crates/evm/fuzz/src/strategies/calldata.rs#L39
 which would use the same strategies generated in https://github.com/foundry-rs/foundry/blob/0a5b22f07ba4f2ddf525089c8ee9cdcb05e44bd9/crates/evm/fuzz/src/strategies/param.rs (as they're actually generated per type)

To make use of proptest API and get current value from strategy we can use a deterministic testrunner (a sample here https://github.com/proptest-rs/proptest/blob/c117b769ba90e0e70c7fa020c29da9aaf89af493/proptest/src/sample.rs#L529-L534), hope that's not overengineered ... What's nice is that we could reuse existing fuzz strategies and only pass the type we want to generate random value from the cheatcode